packages/engine-server/src/drivers/file/storev2.ts

```diff
@@ -430,36 +416,39 @@ export class FileStorage implements DStore {
     };
   }
 
-  async initNotes(): Promise<{ notes: NoteProps[]; errors: DendronError[] }> {
+  async initNotes(): Promise<{
+    errors: IDendronError[];
+  }> {
     const ctx = "initNotes";
     this.logger.info({ ctx, msg: "enter" });
 
     let notesWithLinks: NoteProps[] = [];
-    let errors: DendronError[] = [];
+    let errors: IDendronError<any>[] = [];
     const out = await Promise.all(
       (this.vaults as DVault[]).map(async (vault) => {
         const {
-          notes,
+          notesById,
           cacheUpdates,
           errors: initErrors,
         } = await this._initNotes(vault);
         errors = errors.concat(initErrors);
         notesWithLinks = notesWithLinks.concat(
-          _.filter(notes, (n) => !_.isEmpty(n.links))
+          _.filter(notesById, (n) => !_.isEmpty(n.links))
         );
 
         this.logger.info({
           ctx,
           vault,
-          numEntries: _.size(notes),
+          numEntries: _.size(notesById),
           numCacheUpdates: _.size(cacheUpdates),
         });
 
-        return notes;
+        return notesById;
       })
     );
-    const allNotes = _.flatten(out);
-    if (allNotes.length === 0) {
+    this.notes = Object.assign({}, ...out);
```

Comment:
you can also just use `{...out}` as a more concise verison of this

Reply:
Hmm got type error here. FYI `out` is `NotePropsDict[]` here that I'm trying to merge into one `NotePropsDict`